src/io/ResultWriter.cpp

```diff
@@ -52,11 +52,13 @@ void ResultWriter::init(ParticleContainer * /*particleContainer*/,
 	}
 }
 
-void ResultWriter::endStep(ParticleContainer * /*particleContainer*/, DomainDecompBase *domainDecomp, Domain *domain,
+void ResultWriter::endStep(ParticleContainer *particleContainer, DomainDecompBase *domainDecomp, Domain *domain,
                            unsigned long simstep) {
 
 	// Writing of cavities now handled by CavityWriter
 
+	domain->updateglobalNumMolecules(particleContainer, domainDecomp);
```

Comment:
maybe a comment why this is necessary or why it is not handled elsewhere. Seems strange that this needs to be done from within a plugin?

Reply:
I think, until now, the method is only called in the following:

1. When reading the xml config (`Simulation.cpp:738`)
2. When writing a checkpoint (`Domain.cpp:611`)
3. When a new slab is inserted by the MettDaemon plugin (`MettDeamon.cpp:1084`)

This leads to a wrong number of particles stored in `_globalNumMolecules` (`Domain.cpp`), when plugins like Mirror or MaxCheck delete particles without notice.
Of course, it would be good to call the `updateglobalNumMolecules` method from time to time during the simulation. Maybe you know a good place/time when to call it.